src/gpu/intel/ocl/reusable_softmax.hpp

```diff
@@ -201,6 +201,13 @@ struct reusable_softmax_fwd_t : public gpu_primitive_t {
                 }
             }
 
+            // subgroup block read/write requires the tensor to be
+            // 4/16-byte aligned, respectively
+            const bool is_read_aligned
+                    = (axis_size() * types::data_type_size(src_dt)) % 4 == 0;
+            const bool is_write_aligned
+                    = (axis_size() * types::data_type_size(dst_dt)) % 16 == 0;
```

Comment:
Did you try just adding a runtime branch based on alignment (similar to [this](https://github.com/uxlfoundation/oneDNN/blob/d863a861f3c3e811f3fb0e634463e63cad01de48/src/gpu/intel/ocl/gen9_eltwise.cl#L43-L53))? For memory bound kernels, this conditional check should not be very expensive. We could even add an abstraction for conditional block load to ocl_io.h to make it simple to reuse.

Reply:
Doing a host-side check is in the style of https://github.com/uxlfoundation/oneDNN/blob/d8d1ead904a1db0cf031bcfde88633dd6ae1ce58/src/gpu/intel/ocl/gen9_softmax.cl#L298 . I'm trying to avoid runtime checks as much as possible since that's the penalty to pay for reusable implementations. 

Of course, this could look different, but let's get this bug fixed before making larger scale improvements.